src/BasicRum/Diagram/Builder/RenderType/RenderTypeFactory.php

```diff
@@ -0,0 +1,44 @@
+<?php
```

Comment:
I am looking again at this class and here is my feedback:

- It doesn't look much as a factory that I am familiar with because a factory class usually does the following: Create an object and pass the object to another piece of code that knows what to do with this object. I don't think that we should have build() function for RenderTypeFactory
- We don't need to have checkIfRenderTypeExits() function. This logic could be directly in the constructor.
- I would advice that we do not use a constructor but we implement a static factory explained in this article: https://webmobtuts.com/backend-development/creating-classes-with-php-factory-abstract-factory-simple-and-static-factory-patterns/

Reply:
Hello @cescolab
Thank you for you feedback I'd to love to implement static factory pattern. Meantime could you please explain in your opinion based on what criteria ones should choose when to use static factory and when to choose simple factory.

Thank you

Reply:
There are couple of things that are good and bad depending on the method we would like to choose.

**Simple factory:**

The Simple factory looks more clear in terms what object type will be returned. I would say the code looks more clear when you read the part were a simple factory method is called.

E.g.

```
   public function createBicycle() : Bicycle
    {
        return new Bicycle();
    }
 
    public function createCar() : Car
    {
        return new Car();
    }
```

**Static factory:**

The Static factory implements more logic in its factory method and kind of hides some logic from a developer which could be good or bad depending on the situation. It tries to figure out what object should be created based on a parameter passed though a factory method. In our case you named this method **build**.


I would use a simple factory if I had to call factory methods in many different places in our application where the context is already clear and it's clear which factory function must be called.